rustls/src/crypto/ring/ticketer.rs

```diff
@@ -153,6 +178,15 @@ impl Debug for AeadTicketer {
     }
 }
 
+/// Non-panicking `let (nonce, ciphertext) = ciphertext.split_at(...)`.
+fn try_split_at(slice: &[u8], mid: usize) -> Option<(&[u8], &[u8])> {
```

Comment:
`split_at_checked()` coming... soonish (but will take a while to trickle out to our MSRV).

Reply:
ah, that's great. i went looking for `try_split_at` and didn't find that